airbyte-integrations/connectors/source-notion/source_notion/streams.py

```diff
@@ -298,3 +298,65 @@ def should_retry(self, response: requests.Response) -> bool:
             else:
                 return super().should_retry(response)
         return super().should_retry(response)
+
+
+class Comments(HttpSubStream, IncrementalNotionStream):
+    """
+    Comments Object Docs: https://developers.notion.com/reference/comment-object
+    Comments Endpoint Docs: https://developers.notion.com/reference/retrieve-a-comment
+    """
+
+    http_method = "GET"
+    # We can use the "last edited time" of the parent Page as the cursor field,
+    # since we cannot guarantee the order of comments between pages.
+    cursor_field = "page_last_edited_time"
+
+    def path(self, **kwargs) -> str:
+        return f"comments"
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        block_id = stream_slice.get("block_id", "")
```

Comment:
is there a valid situation where block_id wouldn't be set? or should this throw an error if it can't find block ID?

Reply:
Good point, there's no valid reason that I can think of that block_id wouldn't be properly set, since every parent record's id is fetched directly from the API. I can remove the default argument (it will never trigger and after testing it would have just resulted in a 400 error anyway).